torch_xla/csrc/data_ops.cpp

```diff
@@ -92,6 +92,25 @@ xla::XlaOp BuildView(xla::XlaOp input, absl::Span<const int64_t> output_sizes) {
   return XlaHelpers::DynamicReshape(input, complete_output_sizes);
 }
 
+xla::XlaOp SetDimensionSizes(xla::XlaOp input,
+                             absl::Span<const xla::XlaOp> symbolic_output_sizes,
+                             std::vector<bool> dynamic_dims) {
+  size_t current_output_size_index = 0;
+  size_t symbolic_output_sizes_len = symbolic_output_sizes.size();
+  for (size_t i = 0; i < dynamic_dims.size(); i++) {
+    if (dynamic_dims[i]) {
+      // Dimension i is dynamic
+      XLA_CHECK_LT(current_output_size_index, symbolic_output_sizes_len);
+      input = xla::SetDimensionSize(
+          input, symbolic_output_sizes[current_output_size_index++], i);
```

Comment:
In what scenario `current_output_size_index` and `i` deviate in value?

Reply:
static dimension. `symbolic_output_sizes` only contains dynamic dimensions, where `dynamic_dims` is a boolean array that also includes static dimensions

Reply:
Yup, this [PR](https://github.com/pytorch/xla/pull/3944) covers the context on why we may deviate.

Reply:
@miladm hehe I did warn @JackCaoG it could get confusing (discussion here: https://github.com/pytorch/xla/pull/3889/files#r975882060 ) 

There's quite a bit of logic that tries really hard to separate dynamic dimensions from static ones whereas the only thing we really seem to care about is if there's at least one dynamic dimension (so we need to use `ExpandSymInt::Lower`) assuming of course XLA can optimize `xla::SetDimensionSize(Scalar(X))` *wink* *wink*